src/lib/operators/validate.cpp

```diff
@@ -130,21 +157,28 @@ void Validate::_validate_chunks(const std::shared_ptr<const Table>& in_table, co
       const auto& pos_list_in = *ref_segment_in->pos_list();
       if (pos_list_in.references_single_chunk() && !pos_list_in.empty()) {
         // Fast path - we are looking at a single referenced chunk and thus need to get the MVCC data vector only once.
-
         pos_list_out->guarantee_single_chunk();
 
         const auto referenced_chunk = referenced_table->get_chunk(pos_list_in.common_chunk_id());
         auto mvcc_data = referenced_chunk->get_scoped_mvcc_data_lock();
 
-        for (auto row_id : pos_list_in) {
-          if (opossum::is_row_visible(our_tid, snapshot_commit_id, row_id.chunk_offset, *mvcc_data)) {
-            pos_list_out->emplace_back(row_id);
+        if (check_visibility_on_chunk_level &&
+            is_entire_chunk_visible(referenced_chunk, snapshot_commit_id, mvcc_data)) {
+          // We can simply copy the whole PosList since it is entirely visible. Copying is forbidden for PosLists.
+          // Because of that, we use memcpy here for now.
+          const auto pos_list_size = pos_list_in.size();
+          pos_list_out->resize(pos_list_size);
+          std::memcpy(pos_list_out->data(), pos_list_in.data(), pos_list_in.size() * sizeof(RowID));
```

Comment:
At one point we discussed adding a copy method to the PosList for another issue. Maybe this could be a solution here.